porch/controllers/rootsyncsets/pkg/controllers/rootsyncset/controller.go

```diff
@@ -368,11 +375,15 @@ func (r *RootSyncSetReconciler) pruneWatches(rssnn types.NamespacedName, cluster
 		delete(w.liens, rssnn)
 		// If no other RootSyncSets need the watch, stop it and remove the watcher from the map.
 		if len(w.liens) == 0 {
-			klog.Infof("clusterRef %s is no longer needed, so closing watch", clusterRef.Name)
 			w.cancelFunc()
 			delete(r.watchers, clusterRef)
+			stoppedWatchers = append(stoppedWatchers, clusterRef)
```

Comment:
FYI I think you can delete from a map during iteration, but it does terrify me  :-)

Did you mean to keep the `delete(r.watchers, clusterRef)` line here?